File: src/main/scala/org/biodatagraphdb/alsdb/value/AlsPropertyCategory.scala
package org.biodatagraphdb.alsdb.value

import org.apache.commons.csv.CSVRecord

case class AlsPropertyCategory(category:String, parentCategory:String ){
  val id:String = category
  val isSelfReferential:Boolean = category.equalsIgnoreCase(parentCategory)
}
object AlsPropertyCategory extends ValueTrait {

  def parseCSVRecord(record:CSVRecord):AlsPropertyCategory = {
    new AlsPropertyCategory(
      record.get("category"),
      record.get("parent_category")
    )
  }
}

File: src/main/kotlin/org/biodatagraphdb/alsdb/model/AlsPropertyCategory.kt
package org.biodatagraphdb.alsdb.model

import org.apache.commons.csv.CSVRecord

/**
 * Created by fcriscuo on 2/3/20.
 */

data class AlsPropertyCategory(
        val category:String, val parentCategory:String) {
    val id = category
    val isSelfReferential = category == parentCategory

    companion object : AlsdbModel {
        const val CATEGORY = "category"
        const val PARENT_CATEGORY = "parent_category"
        fun parseCSVRecord(record: CSVRecord): AlsPropertyCategory =
                AlsPropertyCategory(record.get(CATEGORY),
                    record.get(PARENT_CATEGORY))
    }
}